src/connection_manager.c

```diff
@@ -176,6 +176,46 @@ static void set_config_host(qd_server_config_t *config, qd_entity_t* entity)
     assert(config->host);
 }
 
+static char* qd_config_ssl_profile_get_password(qd_config_ssl_profile_t* ssl_profile)
+{
+    char *pw = ssl_profile->ssl_password;
+    if (!pw) return pw;
+
+    /* if the "password" starts with "env:" or "env: " then the remaining
+     * text is the environment variable that contains the password
+     */
+    if (strncmp(pw, "env:", 4) == 0) {
+        char *env = pw + 4;
+        /* skip the space if it is there */
+        if (*env == ' ') ++env;
+
+        const char* passwd = getenv(env);
+        if (passwd) {
+            free(ssl_profile->ssl_password);
+            pw = ssl_profile->ssl_password = strdup(passwd);
```

Comment:
I think your memory management is correct.  This code actually replaces the "env:VAR" text with (a copy of) the contents of the environment variable on the first reference to the ssl_profile.  Subsequent references will then see the contents and won't re-query the environment.
There is one subtle issue with this.  If the password from the environment begins with "env:", the second reference to the ssl_profile will interpret the password as another environment reference.  To prevent this, the copy of the password taken from the environment should be prefixed with "literal:".  Alternatively, you could leave the ssl_profile unmodified and simply return the password from the environment.